scripts/make_typemap/make_typemap.go

```diff
@@ -26,16 +26,26 @@ func snakeCase(camelCase string) string {
 	result := make([]rune, 0)
 	for i, s := range camelCase {
 		tl := strings.ToLower(string(s))
-		if i == 0 {
-			result = append(result, []rune(tl)...)
-			continue
+
+		// Treat acronyms as single-word, e.g. LDAP -> ldap
+		var nextCharCaseChanges bool
+		if i+1 < len(camelCase) {
+			nextChar := camelCase[i+1]
+			// Check if the next character case differs from the current character
+			if (s >= 'a' && s <= 'a' && nextChar >= 'A' && nextChar <= 'Z') || (s >= 'A' && s <= 'Z' && nextChar >= 'a' && nextChar <= 'z') {
```

Comment:
You could simplify this expression with `unicode.IsUpper` and `unicode.IsLower`

https://golang.org/pkg/unicode/